buy_together_app/models.py

```diff
@@ -1,3 +1,42 @@
-# from django.db import models
+from django.db import models
+from django.core.validators import MinLengthValidator
 
-# Create your models here.
+
+class Product(models.Model):
+    qr_code = models.CharField(max_length=32, primary_key=True)
+    product_name = models.CharField(max_length=32, validators=[MinLengthValidator(1)])
+    description = models.TextField(validators=[MinLengthValidator(1)])
+
+    @staticmethod
+    def filter_qr(qr):
+        if qr:
+            return Product.objects.filter(qr_code=qr)
+        return Product.objects.all()
+
+    @staticmethod
+    def filter_name(nm):
+        if nm:
+            return Product.objects.filter(product_name=nm)
+        return Product.objects.all()
+
+    @staticmethod
+    def filter_description(desc):
+        if desc:
+            return Product.objects.filter(description=desc)
+        return Product.objects.all()
+
+    @staticmethod
+    def get_all_products():
+        return Product.objects.all()
+
+    @staticmethod
+    def create_product(prd):
+        if len(prd.product_name) == 0 or len(prd.description) == 0 or len(prd.qr_code) != 32:
```

Comment:
Here we are making sure that we can actually save the product,rather than saving it and getting error from the database
Min to qr was missing ,added

Reply:
What's the idea behind it? avoiding the exception? If so, just wrap it in try/except. Why replicate the logic you already define in the model? Suppose you change it there, you'd need to change it here as well.

Reply:
Yes, i have updated it,i will push all in a moment